week2/MAKEME.md

```diff
@@ -0,0 +1,112 @@
+8. Test your components by importing it into the top level component, which is `<App>`
+
+**Exercise 3: Random Joke Generator**
+
+Never run out of jokes with the Random Joke Generator!
+
+In this exercise we'll be using the following API endpoint: `https://official-joke-api.appspot.com/random_joke`
+
+1. Create 2 functional components: `<RandomJoke>` and `<Joke>`
+2. Inside `<RandomJoke>` create a state variable called `joke` (with initial state `{}`) and state handler `setJoke`
+3. Make use of the `useEffect` hook to make an asynchronous API call to the API endpoint. Put the end result into the state variable using `setJoke`
+4. Now pass the reassigned state variable `joke` to the `<Joke>` component
+5. Inside the `<Joke>` component, display the `setup` and `punchline` attributes in 2 `<p>` tags
+6. Test out your work by importing your components into the top level component, which is `<App>`
+
+## **3. Code along**
+
+In the following tutorial you'll learn how to
+
+- [Build a Calculator with React and Flexbox](https://www.youtube.com/watch?v=KzYUuTiHdiY)
+
+## **4. PROJECT: Hack Your Weather II**
+
+> Make sure you're building on last week's codebase.
+
+Let's continue building on our application! This week you'll replace the application's data source: instead of using a JSON file you'll be using an external API, called [OpenWeatherMap](https://openweathermap.org/). In addition to that, you'll add a `search functionality` that allows a user to type in a city name and retrieve weather data on it!
+
+### 1. Setup
+
+1. Sign up for an API key at [OpenWeatherMap App ID](https://openweathermap.org/appid)
+2. Create a `.env` file in the root of your project folder
+3. Paste the API key inside the `.env` file with the name `OPENWEATHERMAP_API_KEY`
+
+### 2. Requirements
+
+By the end of this week's assignment your application should look similar to this:
+
+![Week 2 Wireframe](../assets/project/week2.png)
+
+Here are the requirements your project needs to fulfill:
+
+- Your form should have `controlled inputs`
+- Make an API call when the button is clicked
+- Display the city + weather information similar to week 1
+- Make a component for the search
+- Incorporate error handling for the API call
+- Show user feedback when there are no cities searched for yet
+
+Hints:
+
+- For the form, use the `onChange` and `value` attributes
+- Take the name of the city to use in the API call
+
+## **SUBMIT YOUR HOMEWORK!**
+
+After you've finished your list it's time to show us what you got! Upload all your files to a forked repository (a copy from the original, which in this case is the [React](https://www.github.com/HackYourFuture/React) repository) using GIT. Then make a pull request to your teacher's forked repository.
+
+If you need a refresher, take a look at the following [guide](../hand-in-homework-guide.md) to see how it's done.
+
+The homework that needs to be submitted is the following:
+
+1. React exercises
+2. Project: Hack Your Weather II
+
```

Comment:
let's also ask them to deploy every homework on netlify/now and add the link to pull request
willhelp in reviewing

Reply:
I'll put it in there.